cmd/dep/ensure.go

```diff
@@ -790,6 +790,10 @@ func checkErrors(m map[string]pkgtree.PackageOrErr, ignore *pkgtree.IgnoredRules
 		}
 	}
 
+	if allowNoGo {
+		noGoErrors = 0
+	}
+
 	// If pkgtree was empty or all dirs lacked any Go code, return an error.
 	if len(m) == 0 || len(m) == noGoErrors {
```

Comment:
the only problem is the `len(m) == 0` case. it's not likely to be reachable under normal circumstances - the directory `Gopkg.toml` is in will always be scanned, so there's always going to be at least one - but it's probably possible to hit it with ignores. so, both checks will need modification.

Reply:
Fixed in 40f0f41d